openpos-client-libs/projects/openpos-client-core-lib/src/lib/core/startup/startup-controller.service.ts

```diff
@@ -0,0 +1,43 @@
+import { Injectable, Injector } from '@angular/core';
+import { startupSequence } from './startup-sequence';
+
+export type StartupControllerState = true | 'in-progress' | false | 'error';
+
+export interface StartupControllerStatus {
+    completed: StartupControllerState;
+    error?: any;
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class StartupController {
+    get status(): StartupControllerStatus {
+        return this._status;
+    }
+
+    private _status: StartupControllerStatus = {
+        completed: false
+    };
+
+    constructor(private _injector: Injector) { }
+
+    async beginStartupSequence(): Promise<StartupControllerStatus> {
+        if (this._status.completed !== false) {
+            throw new Error('startup sequence already initiated');
+        }
+
+        this._status = { completed: 'in-progress' };
+
+        try {
+            await startupSequence.executeTask(this._injector);
+        } catch (e) {
+            this._status = { completed: 'error', error: e };
+            return;
+        }
+
+        this._status = { completed: true };
```

Comment:
Same thing here. Why create a whole new object each time
```suggestion
        this._status.completed = true;
```